addons/binding/org.openhab.binding.velbus/ESH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,656 @@
+			<channel id="CH4" typeId="trigger-channel" />
+			<channel id="CH5" typeId="trigger-channel" />
+			<channel id="CH6" typeId="trigger-channel" />
+			<channel id="CH7" typeId="trigger-channel" />
+		</channels>
+
+        <config-description-ref uri="thing-type:velbus:7channelDevice" />
+	</thing-type>
+
+	<thing-type id="vmbpirm">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>VMBPIRM</label>
+		<description>Mini motion and twilight sensor for recessed or surface mounting</description>
+		<channels>
+			<channel id="CH1" typeId="trigger-channel" />
+			<channel id="CH2" typeId="trigger-channel" />
+			<channel id="CH3" typeId="trigger-channel" />
+			<channel id="CH4" typeId="trigger-channel" />
+			<channel id="CH5" typeId="trigger-channel" />
+			<channel id="CH6" typeId="trigger-channel" />
+			<channel id="CH7" typeId="trigger-channel" />
+		</channels>
+
+        <config-description-ref uri="thing-type:velbus:7channelDevice" />
+	</thing-type>
+
+	<thing-type id="vmbpiro">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="bridge" />
+		</supported-bridge-type-refs>
+
+		<label>VMBPIRO</label>
+		<description>Outdoor motion, twilight and temperature sensor, Theben</description>
+		<channels>
+			<channel id="CH1" typeId="trigger-channel" />
+			<channel id="CH2" typeId="trigger-channel" />
+			<channel id="CH3" typeId="trigger-channel" />
+			<channel id="CH4" typeId="trigger-channel" />
+			<channel id="CH5" typeId="trigger-channel" />
+			<channel id="CH6" typeId="trigger-channel" />
+			<channel id="CH7" typeId="trigger-channel" />
+			<channel id="CH8" typeId="trigger-channel" />
+			<channel id="CH9" typeId="temperature" />
+		</channels>
+
+        <config-description-ref uri="thing-type:velbus:9channelDeviceWithTemperatureSensor" />
+	</thing-type>
+
+	<channel-type id="switch">
+		<item-type>Switch</item-type>
+		<label>Switch</label>
+		<description>Switch control for action in Velbus</description>
+		<category>Switch</category>
+	</channel-type>
+	<channel-type id="brightness">
+		<item-type>Dimmer</item-type>
+		<label>Brightness</label>
+		<description>Brightness control for dimmer action in Velbus</description>
+		<category>DimmableLight</category>
+	</channel-type>
+	<channel-type id="rollershutter">
+		<item-type>Rollershutter</item-type>
+		<label>Rollershutter</label>
+		<description>Rollershutter control for rollershutter action in Velbus</description>
+		<category>Blinds</category>
+	</channel-type>
+	<channel-type id="temperature">
+		<item-type>Number:Temperature</item-type>
+		<label>Temperature</label>
+		<description>Current temperature in degrees celsius</description>
+		<category>Temperature</category>
+		<state readOnly="true" pattern="%.1f °C">
+		</state>
+	</channel-type>
+	<channel-type id="trigger-channel">
+		<kind>trigger</kind>
+		<label>Trigger Channel</label>
+		<event>
+			<options>
+				<option value="PRESSED">pressed</option>
```

Comment:
Are these really the options that you get from the system?
It is a weird mix of the ESH SYSTEM_RAWBUTTON and SYSTEM_BUTTON channels. Usually, you have pressed/release (what we call "raw" events) and from this you can determine through software whether it was a long press, double press or whatever. Or you have this processing done by the button in hardware already, but then you hardly ever get a "released" event.

Reply:
Yes, for each of the events (pressed, released and long pressed) the hardware puts the following message on the bus:
![image](https://user-images.githubusercontent.com/12730069/39469077-cafbbf98-4d36-11e8-8086-8e84afa1473d.png)

For a normal press there are two messages, with last 3 bytes 1,0,0 for the first message and 0,1,0 for the second message.
For a long press there are three messages, with last 3 bytes 1,0,0 for the first message, 0,0,1 for the second message and 0,1,0 for the third message.

Reply:
ok, thanks for the insights!